coloring/custom_color_provider/src/index.ts

```diff
@@ -0,0 +1,118 @@
+import { StructureSelectionQuery } from "molstar/lib/mol-plugin-state/helpers/structure-selection-query";
+import { createRootViewer } from "./common/init";
+import { Bond, StructureElement, StructureProperties } from "molstar/lib/mol-model/structure";
+import { ThemeDataContext } from "molstar/lib/mol-theme/theme";
+import { ParamDefinition as PD } from "molstar/lib/mol-util/param-definition";
+import { ColorTheme } from "molstar/lib/mol-theme/color";
+import { ColorNames } from 'molstar/lib/mol-util/color/names';
+import { MolScriptBuilder as MS } from 'molstar/lib/mol-script/language/builder';
+
+async function init() {
+    // Create viewer
+    const plugin = await createRootViewer();
+
+    // Download mmCIF
+    const fileData = await plugin.builders.data.download(
+        { url: "https://models.rcsb.org/9INS.bcif", isBinary: true }
+    );
+
+    // Load mmCIF and create representation
+    const trajectory = await plugin.builders.structure.parseTrajectory(fileData, "mmcif");
+    const presetStateObjects = await plugin.builders.structure.hierarchy.applyPreset(trajectory, "default");
+
+    if (!presetStateObjects) {
+        throw new Error("Structure not loaded");
+    }
+
+    // The goal here is to color all atoms that have alternate locations with a different color
+    // based on their alt location ID
+
+    // Create an expression using molscript to select all atoms with an alt loc that is not empty
+    const altLocExp = MS.struct.generator.atomGroups({
+        'atom-test': MS.core.rel.neq([MS.struct.atomProperty.macromolecular.label_alt_id(), ''])
+    });
+    // Create a StructureSelectionQuery from the expression to pass it to the component builder
+    const altLocSelectionQuery = StructureSelectionQuery('alt-loc', altLocExp)
+
+    // From the alt-loc selection query, create a component for which we will later create a representation
+    const component = await plugin.builders.structure.tryCreateComponentFromSelection(
+        presetStateObjects.structure,   // The structure state object to create components from
+        altLocSelectionQuery,           // The selection query for the alt-loc atoms
+        "alt-loc"                       // A key to identify the component
+    );
+
+    // If the component failed to be created, throw an error
+    if (!component?.cell) {
+        throw new Error("Failed to create component from selection");
+    }
+
+    // Add the custom color theme to the plugin's color theme registry (only needs to be done once)
+    plugin.representation.structure.themes.colorThemeRegistry.add(BallAndStickAltLocColorThemeProvider);
+
+    // Add a ball-and-stick representation to the component we created
+    // Use our custom color theme to color the atoms in our representation
+    await plugin.builders.structure.representation.addRepresentation(
```

Comment:
you can also pass the providers directly

```
plugin.builders.structure.representation.addRepresentation(
            component.cell,
            {
                type: BallAndStickRepresentationProvider,
                color: UncertaintyColorThemeProvider
            },
        );
```

Reply:
Ok thanks, this makes sense. I wasn’t sure but I was following the schema I was seeing in other examples. I will update this PR in the morning